packages/http-client-csharp/emitter/src/emitter.ts

```diff
@@ -65,7 +66,10 @@ export async function $onEmit(context: EmitContext<NetEmitterOptions>) {
     const tspNamespace = root.Name; // this is the top-level namespace defined in the typespec file, which is actually always different from the namespace of the SDK
     // await program.host.writeFile(outPath, prettierOutput(JSON.stringify(root, null, 2)));
     if (root) {
-      const generatedFolder = resolvePath(outputFolder, "src", "Generated");
+      const isSrcFolder = path.basename(outputFolder) === "src";
```

Comment:
this emitter should not be doing this, it should always assume root.

autorest.csharp should handle both cases.